File: android/app/src/main/java/com/dayliosac/data/model/Mood.kt
package com.dayliosac.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Entité représentant une humeur
 */
@Entity(tableName = "moods")
data class Mood(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    
    val level: Int, // 1-5
    
    val label: String,
    
    val emoji: String? = null,
    
    val color: String? = null
) {
    companion object {
        /**
         * Humeurs par défaut
         */
        fun getDefaultMoods(): List<Mood> = listOf(
            Mood(1, 1, "Très mauvais", "😞", "#FF0000"),
            Mood(2, 2, "Mauvais", "😟", "#FF8800"),
            Mood(3, 3, "Neutre", "😐", "#FFDD00"),
            Mood(4, 4, "Bon", "🙂", "#88FF00"),
            Mood(5, 5, "Excellent", "😊", "#00FF00")
        )
    }
}

File: ios/DaylioSAC/Models/Mood.swift
//
//  Mood.swift
//  DaylioSAC
//
//  Modèle de données pour l'humeur
//

import Foundation

struct Mood: Codable, Identifiable {
    let id: Int64
    let level: Int
    let label: String
    let emoji: String?
    let color: String?
    
    init(
        id: Int64 = 0,
        level: Int,
        label: String,
        emoji: String? = nil,
        color: String? = nil
    ) {
        self.id = id
        self.level = level
        self.label = label
        self.emoji = emoji
        self.color = color
    }
    
    // Humeurs par défaut
    static let defaultMoods: [Mood] = [
        Mood(id: 1, level: 1, label: "Très mauvais", emoji: "😞", color: "#FF0000"),
        Mood(id: 2, level: 2, label: "Mauvais", emoji: "😟", color: "#FF8800"),
        Mood(id: 3, level: 3, label: "Neutre", emoji: "😐", color: "#FFDD00"),
        Mood(id: 4, level: 4, label: "Bon", emoji: "🙂", color: "#88FF00"),
        Mood(id: 5, level: 5, label: "Excellent", emoji: "😊", color: "#00FF00")
    ]
}
